jwst/pipeline/calwebb_image2.py

```diff
@@ -51,8 +50,12 @@ def process(self, input):
 
             # Save result
             suffix = 'cal'
+            if isinstance(input, datamodels.CubeModel):
```

Comment:
This logic isn't working, because `input` is still just a file name at this point - it hasn't been loaded into a data model. If you change this check to use `result` it works fine. calwebb_spec2 may have the same problem.